In [65]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


# Latar Belakang

Sebuah maskapai penerbangan ingin mengetahui seberapa baik pelayanan yang telah diberikan untuk membuat penumpang merasa puas atas pelayanan mereka. Banyak jenis pelayanan yang telah dilakukan oleh maskapai penerbangan ini, namun tidak semua penumpang merasa puas atas pelayanan yang diberikan.  

## Pertanyataan Masalah

Apa yang perlu dilakukan oleh maskapai penerbangan untuk **meningkatkan tingkat kepuasan penumpang terhadap pelayanan yang telah diberikan.**

Sebagai seorang *data analyst*, saya akan mencoba menjawab pertanyaan berikut:

1. **Jenis pelayanan apa saja yang kurang memberikan hasil yang baik terhadap kepuasan penumpang?**

1. **Rekomendasi langkah perbaikan seperti apa yang dapat dilakukan untuk meningkatkan kepuasan penumpang?**

1. **Target peningkatan pelayanan seperti apa yang harus diprioritaskan.**


# Data Understanding and Cleaning
Untuk menjawab pertanyaan di atas, kita akan menganalisa data penumpang yang sudah dikumpulkan oleh perusahaan maskapai penerbangan.

In [66]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import plotly.express as px
from scipy.stats import normaltest, chi2_contingency, mannwhitneyu, ttest_ind
import plotly.graph_objects as go
from plotly.subplots import make_subplots
import plotly.io as pio

import warnings
warnings.filterwarnings("ignore")


In [67]:
df=pd.read_csv('/content/drive/MyDrive/Purwadhika/airline passanger satisfaction.csv')
df

Unnamed: 0      id  Gender      Customer Type  Age   Type of Travel  \
0       0           70172   Male    Loyal Customer     13   Personal Travel   
1       1           5047    Male    disloyal Customer  25   Business travel   
2       2           110028  Female  Loyal Customer     26   Business travel   
3       3           24026   Female  Loyal Customer     25   Business travel   
4       4           119299  Male    Loyal Customer     61   Business travel   
...    ..              ...   ...               ...     ..               ...   
103899  103899      94171   Female  disloyal Customer  23   Business travel   
103900  103900      73097   Male    Loyal Customer     49   Business travel   
103901  103901      68825   Male    disloyal Customer  30   Business travel   
103902  103902      54173   Female  disloyal Customer  22   Business travel   
103903  103903      62567   Male    Loyal Customer     27   Business travel   

           Class  Flight Distance  Inflight wifi service  \
0       Eco Plus  460              3                       
1       Business  235              3                       
2       Business  1142             2                       
3       Business  562              2                       
4       Business  214              3                       
...          ...  ...             ..                       
103899  Eco       192              2                       
103900  Business  2347             4                       
103901  Business  1995             1                       
103902  Eco       1000             1                       
103903  Business  1723             1                       

        Departure/Arrival time convenient  ...  Inflight entertainment  \
0       4                                  ...  5                        
1       2                                  ...  1                        
2       2                                  ...  5                        
3       5                                  ...  2                        
4       3                                  ...  3                        
...    ..                                  ... ..                        
103899  1                                  ...  2                        
103900  4                                  ...  5                        
103901  1                                  ...  4                        
103902  1                                  ...  1                        
103903  3                                  ...  1                        

        On-board service  Leg room service  Baggage handling  Checkin service  \
0       4                 3                 4                 4                 
1       1                 5                 3                 1                 
2       4                 3                 4                 4                 
3       2                 5                 3                 1                 
4       3                 4                 4                 3                 
...    ..                ..                ..                ..                 
103899  3                 1                 4                 2                 
103900  5                 5                 5                 5                 
103901  3                 2                 4                 5                 
103902  4                 5                 1                 5                 
103903  1                 1                 4                 4                 

        Inflight service  Cleanliness  Departure Delay in Minutes  \
0       5                 5            25                           
1       4                 1            1                            
2       4                 5            0                            
3       4                 2            11                           
4       3                 3            0                            
...    ..                ..           ..                      

Dalam dataset yang telah diberikan, ada beberapa informasi yang telah didapatkan. Beberapa informasi tersebut dinyatakan dalam beberapa kolom berikut ini:

- ID: Nomor ID customer.
- Gender: Jenis kelamin customer.

- Customer Type: Jenis customer customer loyal atau baru. Terbagi menjadi dua jenis:
  - Loyal customer.
  - Disloyal customer atau customer baru.

- Age: Usia customer.

- Type of Travel: Tujuan perjalan yang dilakukan oleh customer. Terbagi menjadi dua jenis tujuan perjalanan:
  - Personal Travel.
  - Business Travel.

- Class: Jenis kelas maskapai penerbangan yang digunakan oleh customer. Terbagi menjadi tiga jenis kelas maskapai penerbangan:
  - Eco
  - Eco Plus.
  - Business.

- Flight Distance: Jarak penerbangan masing-masing customer.

- Inflight WiFi service	: Tingkat kepuasan customer terhadap pelayanan WiFi yang diberikan maskapai penerbangan kepada customer.

- Departure/Arrival time convenient: Tingkat kepuasan customer terhadap ketepatan waktu penerbangan.

- Ease of Online Booking: Tingkat kepuasan customer terhadap kemudahan dalam melakukan booking atau pemesanan secara online.

- Gate Location: Tingkat kepuasan customer dengan pelayanan gate dan lokasi gate pesawat.

- Food and drink: Tingkat kepuasan customer dengan pelayanan makanan dan minuman yang diberikan maskapai penerbangan.

- Online boarding: Tingkat kepuasan customer dengan pelayanan online boarding yang diberikan maskapai penerbangan.

- Seat comfort: Tingkat kepuasan customer terhadap kenyamanan kursi saat berada di pesawat.

- Inflight entertainment: Tingkat kepuasan customer terhadap layanan hiburan selama berada di pesawat.

- On-board service: Tingkat kepuasan customer terhadap pelayanan on boarding maskapai penerbangan.

- Leg room service: Tingkat kepuasan customer terhadap kenyamanan ruang kaki selama berada di pesawat.

- Baggage handling: Tingkat kepuasan customer terhadap pelayanan barang yang diberikan maskapai penerbangan.

- Checkin service: Tingkat kepuasan customer terhadap pelayanan checkin yang diberikan maskapai penerbangan.

- Inflight service: Tingkat kepuasan customer terhadap pelayanan maskapai saat berada di pesawat.

- Cleanliness: Tingkat kepuasan customer terhadap kebersihan pesawat.

- Departure Delay in Minutes: Waktu delay atau berapa lama waktu tunggu pesawat tiba.

- Arrival Delay in Minutes: Waktu delay atau berapa lama waktu keterlambatan pesawat tiba di tujuan.

- Satisfaction: Tingkat kepuasan customer terhadap pelayanan maskapai penerbangan. Dinyatakan dalam dua jenis:
  - Satisfied.
  - Neutral or dissatisfied.


## Data Understanding
Sebelum masuk ke dalam analisis, kita perlu mengenal dataset kita lebih jauh dalam tahapan *data understanding*. 

Dalam proses ini, kita akan tahu seperti apa data yang kita miliki, anomali-anomali apa saja yang terdapat di dalam dataset kita dan perlu ditangani dalam tahapan *data cleaning*. 

Setiap penangan anomali yang dilakukan, akan disertai dengan justifikasi langkah yang diambil, baik secara *domain knowledge* maupun secara statistik.

Pertama, mari kita lihat informasi umum dari dataset airline passanger satisfaction.

### Jenis Data

In [68]:
print(f'Jumlah baris dan kolom di dataset airline passanger satisfaction adalah {df.shape}(Baris, Kolom)\n')

print('Jenis data yang ada dalam dataset airline passanger satisfaction \n')
df.info()

Jumlah baris dan kolom di dataset airline passanger satisfaction adalah (103904, 25)(Baris, Kolom)

Jenis data yang ada dalam dataset airline passanger satisfaction 

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 103904 entries, 0 to 103903
Data columns (total 25 columns):
 #   Column                             Non-Null Count   Dtype  
---  ------                             --------------   -----  
 0   Unnamed: 0                         103904 non-null  int64  
 1   id                                 103904 non-null  int64  
 2   Gender                             103904 non-null  object 
 3   Customer Type                      103904 non-null  object 
 4   Age                                103904 non-null  int64  
 5   Type of Travel                     103904 non-null  object 
 6   Class                              103904 non-null  object 
 7   Flight Distance                    103904 non-null  int64  
 8   Inflight wifi service              103904 non-null  int64  
 9   De

### Statistic Descriptive

In [69]:
print('Informasi statistic descriptive dataset airline passanger satisfaction\n')
display(df.describe(), df.describe(include='object'))

Informasi statistic descriptive dataset airline passanger satisfaction



Unnamed: 0             id            Age  Flight Distance  \
count  103904.000000  103904.000000  103904.000000  103904.000000     
mean   51951.500000   64924.210502   39.379706      1189.448375       
std    29994.645522   37463.812252   15.114964      997.147281        
min    0.000000       1.000000       7.000000       31.000000         
25%    25975.750000   32533.750000   27.000000      414.000000        
50%    51951.500000   64856.500000   40.000000      843.000000        
75%    77927.250000   97368.250000   51.000000      1743.000000       
max    103903.000000  129880.000000  85.000000      4983.000000       

       Inflight wifi service  Departure/Arrival time convenient  \
count  103904.000000          103904.000000                       
mean   2.729683               3.060296                            
std    1.327829               1.525075                            
min    0.000000               0.000000                            
25%    2.000000               2.000000                            
50%    3.000000               3.000000                            
75%    4.000000               4.000000                            
max    5.000000               5.000000                            

       Ease of Online booking  Gate location  Food and drink  Online boarding  \
count  103904.000000           103904.000000  103904.000000   103904.000000     
mean   2.756901                2.976883       3.202129        3.250375          
std    1.398929                1.277621       1.329533        1.349509          
min    0.000000                0.000000       0.000000        0.000000          
25%    2.000000                2.000000       2.000000        2.000000          
50%    3.000000                3.000000       3.000000        3.000000          
75%    4.000000                4.000000       4.000000        4.000000          
max    5.000000                5.000000       5.000000        5.000000          

        Seat comfort  Inflight entertainment  On-board service  \
count  103904.000000  103904.000000           103904.000000      
mean   3.439396       3.358158                3.382363           
std    1.319088       1.332991                1.288354           
min    0.000000       0.000000                0.000000           
25%    2.000000       2.000000                2.000000           
50%    4.000000       4.000000                4.000000           
75%    5.000000       4.000000                4.000000           
max    5.000000       5.000000                5.000000           

       Leg room service  Baggage handling  Checkin service  Inflight service  \
count  103904.000000     103904.000000     103904.000000    103904.000000      
mean   3.351055          3.631833          3.304290         3.640428           
std    1.315605          1.180903          1.265396         1.175663           
min    0.000000          1.000000          0.000000         0.000000           
25%    2.000000          3.000000          3.000000         3.000000           
50%    4.000000          4.000000          3.000000         4.000000           
75%    4.000000          5.000000          4.000000         5.000000           
max    5.000000          5.000000          5.000000         5.000000           

         Cleanliness  Departure Delay in Minutes  Arrival Delay in Minutes  
count  103904.000000  103904.000000               103594.000000             
mean   3.286351       14.815618                   15.178678                 
std    1.312273       38.230901                   38.698682                 
min    0.000000       0.000000                    0.000000                  
25%    2.000000       0.000000                    0.000000                  
50%    3.000000       0.000000                    0.000000                  
75%    4.000000       12.000000                   13.000000                 
max    5.000000       1592.000000                 1584.000000

Gender   Customer Type   Type of Travel     Class  \
count   103904  103904          103904           103904     
unique  2       2               2                3          
top     Female  Loyal Customer  Business travel  Business   
freq    52727   84923           71655            49665      

                   satisfaction  
count   103904                   
unique  2                        
top     neutral or dissatisfied  
freq    58879

### Data Exploring

In [70]:
print('Gambaran umum isi dataset airline passanger satisfaction\n')

pd.set_option('display.max_colwidth', -1)
# data uni di tiap kolom
listItem = []
for col in df.columns :
    listItem.append( [col, df[col].nunique(), df[col].unique()])

tabel1Desc = pd.DataFrame(columns=['Column Name', 'Number of Unique', 'Unique Sample'],
                     data=listItem)
tabel1Desc

Gambaran umum isi dataset airline passanger satisfaction



Column Name  Number of Unique  \
0   Unnamed: 0                         103904             
1   id                                 103904             
2   Gender                             2                  
3   Customer Type                      2                  
4   Age                                75                 
5   Type of Travel                     2                  
6   Class                              3                  
7   Flight Distance                    3802               
8   Inflight wifi service              6                  
9   Departure/Arrival time convenient  6                  
10  Ease of Online booking             6                  
11  Gate location                      6                  
12  Food and drink                     6                  
13  Online boarding                    6                  
14  Seat comfort                       6                  
15  Inflight entertainment             6                  
16  On-board service                   6                  
17  Leg room service                   6                  
18  Baggage handling                   5                  
19  Checkin service                    6                  
20  Inflight service                   6                  
21  Cleanliness                        6                  
22  Departure Delay in Minutes         446                
23  Arrival Delay in Minutes           455                
24  satisfaction                       2                  

                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              Unique Sample  
0   [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 50, 51, 52, 53, 54, 55, 56, 57, 58, 59, 60, 61, 62, 63, 64, 65, 66, 67, 68, 69, 70, 71, 72, 73, 74, 75, 76, 77, 78, 79, 80, 81, 82, 83, 84, 85, 86, 87, 88, 89, 90, 91, 92, 93, 94, 95, 96, 97, 98, 99, ...]                                                                                                                                                                                                                                                                                                                                              
1   [70172, 5047, 110028, 24026, 119299, 111157, 82113, 96462, 79485, 65725, 34991, 51412, 98628, 83502, 95789, 100580, 71142, 127461, 70354, 66246, 39076, 22434, 43510, 114090, 105420, 102956, 18510, 14925, 118319, 75460, 48492, 27809, 70594, 30089, 58779, 79659, 110293, 48014, 96517, 64685, 64138, 60373, 14849, 28319, 103012, 124114, 107794, 81983, 53164, 82042, 72127, 113046, 1050, 54886, 125918, 7467, 78972, 126625, 114501, 114813, 32167, 68001, 117052, 8661, 57165, 78610, 51058, 92027, 115534, 9261, 61033, 75930, 112142, 24518, 112524, 101639, 85018, 106922, 32564, 92242, 73302, 42267, 121358, 65655, 88062, 91330, 22031, 27921, 51947, 40017, 76392, 118486, 6012, 30183, 41652, 15138, 123589, 114534, 93076, 96963, ...]  
2   [Male, Female]                                                                                                                                                                                                                                                                               

### Missing Value

In [71]:
# Jumlah Missing Value

df.isna().sum()

Unnamed: 0                           0  
id                                   0  
Gender                               0  
Customer Type                        0  
Age                                  0  
Type of Travel                       0  
Class                                0  
Flight Distance                      0  
Inflight wifi service                0  
Departure/Arrival time convenient    0  
Ease of Online booking               0  
Gate location                        0  
Food and drink                       0  
Online boarding                      0  
Seat comfort                         0  
Inflight entertainment               0  
On-board service                     0  
Leg room service                     0  
Baggage handling                     0  
Checkin service                      0  
Inflight service                     0  
Cleanliness                          0  
Departure Delay in Minutes           0  
Arrival Delay in Minutes             310
satisfaction    

In [72]:
# Persentase Missing Value

df.isna().sum()/df.shape[0]*100

Unnamed: 0                           0.000000
id                                   0.000000
Gender                               0.000000
Customer Type                        0.000000
Age                                  0.000000
Type of Travel                       0.000000
Class                                0.000000
Flight Distance                      0.000000
Inflight wifi service                0.000000
Departure/Arrival time convenient    0.000000
Ease of Online booking               0.000000
Gate location                        0.000000
Food and drink                       0.000000
Online boarding                      0.000000
Seat comfort                         0.000000
Inflight entertainment               0.000000
On-board service                     0.000000
Leg room service                     0.000000
Baggage handling                     0.000000
Checkin service                      0.000000
Inflight service                     0.000000
Cleanliness                       

## Kesimpulan

Berdasarkan beberapa data yang telah ditampilkan sebelumnya, berikut ini adalah beberapa informasi yang dapat kita simpulkan:

- Dataset airline passanger satisfaction memiliki **25 kolom dan 103.904 baris**

- Jenis-jenis data bisa dilihat pada bagian 'Jenis Data' untuk masing-masing kolom yang ada. Dimana dalam dataset airline passanger satisfaction terdapat jenis data interger, float, dan string.  


- Missing value atau data yang hilang hanya berada pada kolom Arrival Delay in Minutes. 
  - Jumlah missing value dalam kolom tersebut adalah 310 dari 103.904, atau sebesar 0.2%

- Beberapa data yang terkait tingkat kepuasan, dibuat dalam bentuk survey dengan jenis data interger, nantinya data ini akan diubah dalam bentuk objek atau string dengan rentang sebagai berikut:
  - 0: Netral
  - 1: Paling tidak puas
  - 2: Tidak puas
  - 3: Puas
  - 4: Lumayan puas
  - 5: Paling puas

- Untuk data pada kolom 'Baggage handling' survey tingkat kepuasan dibuat dalam rentang sebagai berikut:
  - 1: Paling tidak puas
  - 2: Tidak puas
  - 3: Puas
  - 4: Lumayan puas
  - 5: Paling puas

## Data Cleaning

### Mengubah Missing Value

In [73]:
df.loc[df['Arrival Delay in Minutes'].isnull(),'Arrival Delay in Minutes'] = 0

In [74]:
df['Arrival Delay in Minutes'].isnull().sum()

0

In [75]:
df.isna().sum()

Unnamed: 0                           0
id                                   0
Gender                               0
Customer Type                        0
Age                                  0
Type of Travel                       0
Class                                0
Flight Distance                      0
Inflight wifi service                0
Departure/Arrival time convenient    0
Ease of Online booking               0
Gate location                        0
Food and drink                       0
Online boarding                      0
Seat comfort                         0
Inflight entertainment               0
On-board service                     0
Leg room service                     0
Baggage handling                     0
Checkin service                      0
Inflight service                     0
Cleanliness                          0
Departure Delay in Minutes           0
Arrival Delay in Minutes             0
satisfaction                         0
dtype: int64

### Mengubah Hasil Survey

In [76]:
hasil_survey = {
    0: 'Netral',
    1: 'Sangat Tidak Puas',
    2: 'Tidak Puas',
    3: 'Puas',
    4: 'Lumayan Puas',
    5: 'Sangat Puas',
}

In [77]:
for i in df.columns[8:22]:
  df[i].replace(hasil_survey, inplace = True)

In [78]:
df

Unnamed: 0      id  Gender      Customer Type  Age   Type of Travel  \
0       0           70172   Male    Loyal Customer     13   Personal Travel   
1       1           5047    Male    disloyal Customer  25   Business travel   
2       2           110028  Female  Loyal Customer     26   Business travel   
3       3           24026   Female  Loyal Customer     25   Business travel   
4       4           119299  Male    Loyal Customer     61   Business travel   
...    ..              ...   ...               ...     ..               ...   
103899  103899      94171   Female  disloyal Customer  23   Business travel   
103900  103900      73097   Male    Loyal Customer     49   Business travel   
103901  103901      68825   Male    disloyal Customer  30   Business travel   
103902  103902      54173   Female  disloyal Customer  22   Business travel   
103903  103903      62567   Male    Loyal Customer     27   Business travel   

           Class  Flight Distance Inflight wifi service  \
0       Eco Plus  460              Puas                   
1       Business  235              Puas                   
2       Business  1142             Tidak Puas             
3       Business  562              Tidak Puas             
4       Business  214              Puas                   
...          ...  ...               ...                   
103899  Eco       192              Tidak Puas             
103900  Business  2347             Lumayan Puas           
103901  Business  1995             Sangat Tidak Puas      
103902  Eco       1000             Sangat Tidak Puas      
103903  Business  1723             Sangat Tidak Puas      

       Departure/Arrival time convenient  ... Inflight entertainment  \
0       Lumayan Puas                      ...  Sangat Puas             
1       Tidak Puas                        ...  Sangat Tidak Puas       
2       Tidak Puas                        ...  Sangat Puas             
3       Sangat Puas                       ...  Tidak Puas              
4       Puas                              ...  Puas                    
...      ...                              ...   ...                    
103899  Sangat Tidak Puas                 ...  Tidak Puas              
103900  Lumayan Puas                      ...  Sangat Puas             
103901  Sangat Tidak Puas                 ...  Lumayan Puas            
103902  Sangat Tidak Puas                 ...  Sangat Tidak Puas       
103903  Puas                              ...  Sangat Tidak Puas       

         On-board service   Leg room service   Baggage handling  \
0       Lumayan Puas       Puas               Lumayan Puas        
1       Sangat Tidak Puas  Sangat Puas        Puas                
2       Lumayan Puas       Puas               Lumayan Puas        
3       Tidak Puas         Sangat Puas        Puas                
4       Puas               Lumayan Puas       Lumayan Puas        
...      ...                        ...                ...        
103899  Puas               Sangat Tidak Puas  Lumayan Puas        
103900  Sangat Puas        Sangat Puas        Sangat Puas         
103901  Puas               Tidak Puas         Lumayan Puas        
103902  Lumayan Puas       Sangat Puas        Sangat Tidak Puas   
103903  Sangat Tidak Puas  Sangat Tidak Puas  Lumayan Puas        

          Checkin service Inflight service        Cleanliness  \
0       Lumayan Puas       Sangat Puas      Sangat Puas         
1       Sangat Tidak Puas  Lumayan Puas     Sangat Tidak Puas   
2       Lumayan Puas       Lumayan Puas     Sangat Puas         
3       Sangat Tidak Puas  Lumayan Puas     Tidak Puas          
4       Puas               Puas             Puas                
...      ...                ...              ...                
103899  Tidak Puas         Puas             Tidak Puas          
103900  Sangat Puas        Sangat Puas      Lumayan Puas        
103901  Sangat Puas        Sangat Puas      Lumayan Puas        
103902  Sangat Puas        Lum

## Keterangan

- Missing Value
  - Pada bagian missing value, data pada kolom Arrival Delay in Minutes yang bernilai 0.0 memiliki arti tidak adanya keterlambatan pesawat tiba ditujuan.
  - Karena memiliki nilai 0.0, maka data terbaca NaN atau dianggap kosong.
  - Data kosong tersebut kemudian diisi dengan nilai 0. 
Missing Value pada kolom Arrival Delay in Minutes adalah bernilai 0.0 atau tidak ada keterlambatan.

- Mengubah Data
  - Data pada kolom Arrival Delay in Minutes diubah menjadi data dengan tipe interger karena data yang ditampilkan hanya dalam hitungan total menit saja.

- Mengubah Hasil Survey
  - Kolom yang memiliki data hasil survey berupa angka 0 sampai 5 diubah menjadi hasil survey sebagai berikut:
    - 0: Netral
    - 1: Sangat Tidak Puas
    - 2: Tidak Puas
    - 3: Puas
    - 4: Lumayan Puas
    - 5: Sangat Puas


# Data Analysis

## Perbandingan Penumpang Puas dan Tidak Puas

### Jumlah dan Pesentase Perbandingan

In [79]:
# Tabel jumlah penumpang puas dan tidak puas
df_satisfaction = df.groupby(["satisfaction"]).count()["id"].reset_index()
print('Perbandingan Tingkat Kepuasan \n')
df_satisfaction

Perbandingan Tingkat Kepuasan 



satisfaction     id
0  neutral or dissatisfied  58879
1  satisfied                45025

In [80]:
# Jumlah yang netral dan tidak puas

dissatisfied = df[df['satisfaction']=='neutral or dissatisfied']['id'].count()
print('Number of passengers with neutral or dissatisfied survey results: ',dissatisfied)

Number of passengers with neutral or dissatisfied survey results:  58879


In [81]:
# Diagram Jumlah Penumpah Puas dan Tidak Puas
fig = px.bar(df_satisfaction,
             x = "satisfaction",
             y = "id",
             title = "Diagram of Neutral or Dissatisfied and Satisfied Passengers",
             labels={
                 "satisfaction": "Satisfaction Level",
                 "id": "Total Passengers"
             },
             text="id")
fig.show()

# Persentase Perbandingan Jumlah Penumpah Puas dan Tidak Puas
fig = px.pie(df_satisfaction,
             names = "satisfaction", 
             values= "id",
             title = "Percentage of Neutral or Dissatisfied and Satisfied Passengers")
fig.show()

#### Kesimpulan

- Berdasarkan data tersebut, sebanyak 58.879 penumpang atau sebesar 56.7% dari total keseluruhan penumpang merasa tidak puas terhadap layanan yang diberikan.

## Berdasarkan Kategori

In [82]:
# memisahkan data berdasarkan tingkat kepuasan
df_satisfied = df[df['satisfaction'] == 'satisfied']
df_neutral_disatisfied = df[df['satisfaction'] != 'satisfied']

### Gender

In [83]:
# menghitung jumlah penumpang "Satisfied" dan "Neutral or Disatisfied" berdasarkan gender
df_gender_satisfied = df_satisfied.groupby(['Gender', 'satisfaction']).count()['id'].reset_index()
df_gender_neutral_disatisfied = df_neutral_disatisfied.groupby(['Gender', 'satisfaction']).count()['id'].reset_index()

# menggabungkan kedua dataframe
df_gender = pd.concat([df_gender_satisfied, df_gender_neutral_disatisfied], ignore_index=True)

# menampilkan dataframe
print('Perbandingan Tingkat Kepuasan Berdasarkan Gender \n')
print(df_gender)

Perbandingan Tingkat Kepuasan Berdasarkan Gender 

   Gender             satisfaction     id
0  Female  satisfied                22534
1  Male    satisfied                22491
2  Female  neutral or dissatisfied  30193
3  Male    neutral or dissatisfied  28686


In [84]:
# Diagram tingkat kepuasan berdasarkan gender
fig = px.bar(df_gender, 
             x='Gender', 
             y='id', 
             color='satisfaction', 
             barmode='stack',
             title='Number Comparison of Passengers Based on Gender and Level of Satisfaction',
             labels={
                 "satisfaction": "Satisfaction Level",
                 "id": "Total Passengers"
             },
             text="id")
fig.show()

#### Kesimpulan

- Berdasarkan data tersebut, jumlah penumpang yang tidak puas sedikit lebih didominasi oleh perempuan dengan jumlah 30.193 penumpang.

### Customer Type

In [85]:
# menghitung jumlah penumpang "Satisfied" dan "Neutral or Disatisfied" berdasarkan customer type
df_customer_type_satisfied = df_satisfied.groupby(['Customer Type', 'satisfaction']).count()['id'].reset_index()
df_customer_type_disatisfied = df_neutral_disatisfied.groupby(['Customer Type', 'satisfaction']).count()['id'].reset_index()

# menggabungkan kedua dataframe
df_customer_type = pd.concat([df_customer_type_satisfied, df_customer_type_disatisfied], ignore_index=True)

# menampilkan dataframe
print('Perbandingan Tingkat Kepuasan Berdasarkan Customer Type \n')
print(df_customer_type)

Perbandingan Tingkat Kepuasan Berdasarkan Customer Type 

       Customer Type             satisfaction     id
0  Loyal Customer     satisfied                40533
1  disloyal Customer  satisfied                4492 
2  Loyal Customer     neutral or dissatisfied  44390
3  disloyal Customer  neutral or dissatisfied  14489


In [86]:
# Diagram tingkat kepuasan berdasarkan customer type 
fig = px.bar(df_customer_type, 
             x='Customer Type', 
             y='id', 
             color='satisfaction', 
             barmode='stack',
             title='Number Comparison of Passengers Based on Customer Type and Level of Satisfaction',
             labels={
                 "id": "Total Passengers"
             },
             text="id")
fig.show()

#### Kesimpulan

- Berdasarkan data tersebut, dapat dilihat bahwa terjadi perbandingan yang sangat tinggi antara penumpang puas dengan penumpang yang tidak puas, dari tipe penumpang disloyal customer atau penumpang baru. 

- Dapat disimpulkan juga bahwa pelayanan yang dilakukan dalam beberapa waktu terakhir ini tidak memberikan kepuasan kepada penumpang.

### Age

In [87]:
# menghitung jumlah penumpang "Satisfied" dan "Neutral or Disatisfied" berdasarkan usia
df_age_satisfied = df_satisfied.groupby(['Age', 'satisfaction']).count()['id'].reset_index()
df_age_neutral_disatisfied = df_neutral_disatisfied.groupby(['Age', 'satisfaction']).count()['id'].reset_index()

# menggabungkan kedua dataframe
df_age = pd.concat([df_age_satisfied, df_age_neutral_disatisfied], ignore_index=True)

# menampilkan dataframe
print('Perbandingan Tingkat Kepuasan Berdasarkan Usia \n')
print(df_age)

Perbandingan Tingkat Kepuasan Berdasarkan Usia 

     Age             satisfaction   id
0    7    satisfied                56 
1    8    satisfied                69 
2    9    satisfied                95 
3    10   satisfied                90 
4    11   satisfied                109
..   ..         ...                ...
145  77   neutral or dissatisfied  71 
146  78   neutral or dissatisfied  22 
147  79   neutral or dissatisfied  30 
148  80   neutral or dissatisfied  63 
149  85   neutral or dissatisfied  14 

[150 rows x 3 columns]


In [88]:
# Diagram tingkat kepuasan berdasarkan usia
fig = px.bar(df_age, 
             x='Age', 
             y='id', 
             color='satisfaction', 
             barmode='stack',
             title='Number Comparison of Passengers Based on Age and Level of Satisfaction',
             labels={
                 "id": "Total Passengers"
             },
             text="id")
fig.show()

#### Kesimpulan

- Berdasarkan data tersebut, dapat dilihat bahwa mayoritas penumpang yang tidak puas berada pada rentang usia mulai dari 20 tahun sampai 60 tahun, dengan usia tertinggi berada pada usia 25 dan 39 tahun.

### Type of Travel

In [89]:
# menghitung jumlah penumpang "Satisfied" dan "Neutral or Disatisfied" berdasarkan tipe perjalanan
df_type_of_travel_satisfied = df_satisfied.groupby(['Type of Travel', 'satisfaction']).count()['id'].reset_index()
df_type_of_travel_neutral_disatisfied = df_neutral_disatisfied.groupby(['Type of Travel', 'satisfaction']).count()['id'].reset_index()

# menggabungkan kedua dataframe
df_type_of_travel = pd.concat([df_type_of_travel_satisfied, df_type_of_travel_neutral_disatisfied], ignore_index=True)

# menampilkan dataframe
print('Perbandingan Tingkat Kepuasan Berdasarkan Tipe Perjalanan \n')
print(df_type_of_travel)

Perbandingan Tingkat Kepuasan Berdasarkan Tipe Perjalanan 

    Type of Travel             satisfaction     id
0  Business travel  satisfied                41746
1  Personal Travel  satisfied                3279 
2  Business travel  neutral or dissatisfied  29909
3  Personal Travel  neutral or dissatisfied  28970


In [90]:
# Diagram tingkat kepuasan berdasarkan type of travel
fig = px.bar(df_type_of_travel, 
             x='Type of Travel', 
             y='id', 
             color='satisfaction', 
             barmode='stack',
             title='Number Comparison of Passengers Based on Type of Travel and Level of Satisfaction',
            labels={
                 "id": "Total Passengers"
             },
             text="id")
fig.show()

#### Kesimpulan

- Berdasarkan data tersebut, dapat dilihat bahwa terjadi perbandingan yang sangat tinggi antara penumpang puas dengan penumpang yang tidak puas, dari tipe perjalanan personal travel.

- Dapat disimpulkan juga bahwa pelayanan terhadap penumpang dengan tipe perjalanan personal travel perlu segera ditingkatkan.

### Class

In [91]:
# menghitung jumlah penumpang "Satisfied" dan "Neutral or Disatisfied" berdasarkan kelas maskapai
df_class_satisfied = df_satisfied.groupby(['Class', 'satisfaction']).count()['id'].reset_index()
df_class_neutral_disatisfied = df_neutral_disatisfied.groupby(['Class', 'satisfaction']).count()['id'].reset_index()

# menggabungkan kedua dataframe
df_class = pd.concat([df_class_satisfied, df_class_neutral_disatisfied], ignore_index=True)

# menampilkan dataframe
print('Perbandingan Tingkat Kepuasan Berdasarkan Kelas Maskapai \n')
print(df_class)

Perbandingan Tingkat Kepuasan Berdasarkan Kelas Maskapai 

      Class             satisfaction     id
0  Business  satisfied                34480
1  Eco       satisfied                8701 
2  Eco Plus  satisfied                1844 
3  Business  neutral or dissatisfied  15185
4  Eco       neutral or dissatisfied  38044
5  Eco Plus  neutral or dissatisfied  5650 


In [92]:
# Diagram tingkat kepuasan berdasarkan kelas maskapai
fig = px.bar(df_class, 
             x='Class', 
             y='id', 
             color='satisfaction', 
             barmode='stack',
             title='Number Comparison of Passengers Based on Class and Level of Satisfaction',
            labels={
                 "id": "Total Passengers"
             },
             text="id")
fig.show()

#### Kesimpulan

- Berdasarkan data tersebut, dapat dilihat bahwa terjadi perbandingan yang sangat tinggi antara penumpang puas dengan penumpang yang tidak puas, dari kelas penerbangan Eco.

- Dapat disimpulkan juga bahwa kelas penerbangan Eco adalah kelas penerbangan yang paling tidak memberikan kepuasan kepada penumpang untuk saat ini.


## Detail Hasil Survey Pada Setiap Layanan

### Inflight wifi service

In [93]:
df_inflight_wifi_service = pd.DataFrame(df['Inflight wifi service'].value_counts()).reset_index()

print('Hasil Survey Kepuasan Pada Layanan Inflight WiFi Services \n')
print(df_inflight_wifi_service)

Hasil Survey Kepuasan Pada Layanan Inflight WiFi Services 

               index  Inflight wifi service
0  Puas               25868                
1  Tidak Puas         25830                
2  Lumayan Puas       19794                
3  Sangat Tidak Puas  17840                
4  Sangat Puas        11469                
5  Netral             3103                 


In [94]:
# Menampilkan barplot hasil survey inflight wifi services
fig = px.bar(df_inflight_wifi_service,
             x="index", 
             y="Inflight wifi service",
             title="Passenger Satisfaction on Inflight wifi service",
             labels={
                 "Inflight wifi service" : "Total Passengers",
                 "index": "Survey Results"
             },
             text="Inflight wifi service",
             category_orders={"index": [
                 "Netral", 
                 "Sangat Tidak Puas", 
                 "Tidak Puas", 
                 "Puas", 
                 "Lumayan Puas", 
                 "Sangat Puas"]})
fig.show()

# Menampilkan pie chart persentase hasil survey inflight wifi services
fig = px.pie(df_inflight_wifi_service,
             names = "index", 
             values= "Inflight wifi service",
             title = "Percentage of Passenger Satisfaction on Inflight wifi service")
fig.show()

### Departure/Arrival time convenient

In [95]:
df_departure_arrival_time_convenient = pd.DataFrame(df['Departure/Arrival time convenient'].value_counts()).reset_index()

print('Hasil Survey Kepuasan Pada Layanan Departure/Arrival time convenient \n')
print(df_departure_arrival_time_convenient)

Hasil Survey Kepuasan Pada Layanan Departure/Arrival time convenient 

               index  Departure/Arrival time convenient
0  Lumayan Puas       25546                            
1  Sangat Puas        22403                            
2  Puas               17966                            
3  Tidak Puas         17191                            
4  Sangat Tidak Puas  15498                            
5  Netral             5300                             


In [96]:
# Menampilkan barplot hasil survey Departure/Arrival time convenient
fig = px.bar(df_departure_arrival_time_convenient,
             x="index", 
             y="Departure/Arrival time convenient",
             title="Passenger Satisfaction on Departure/Arrival time convenient",
             labels={
                 "Departure/Arrival time convenient" : "Total Passengers",
                 "index": "Survey Results"
             },
             text="Departure/Arrival time convenient",
             category_orders={"index": [
                 "Netral", 
                 "Sangat Tidak Puas", 
                 "Tidak Puas", 
                 "Puas", 
                 "Lumayan Puas", 
                 "Sangat Puas"]})
fig.show()

# Menampilkan pie chart persentase hasil survey Departure/Arrival time convenient
fig = px.pie(df_departure_arrival_time_convenient,
             names = "index", 
             values= "Departure/Arrival time convenient",
             title = "Percentage of Passenger Satisfaction on Departure/Arrival time convenient")
fig.show()

### Ease of Online booking

In [97]:
df_ease_of_online_booking = pd.DataFrame(df['Ease of Online booking'].value_counts()).reset_index()

print('Hasil Survey Kepuasan Pada Layanan Ease of Online booking \n')
print(df_ease_of_online_booking)

Hasil Survey Kepuasan Pada Layanan Ease of Online booking 

               index  Ease of Online booking
0  Puas               24449                 
1  Tidak Puas         24021                 
2  Lumayan Puas       19571                 
3  Sangat Tidak Puas  17525                 
4  Sangat Puas        13851                 
5  Netral             4487                  


In [98]:
# Menampilkan barplot hasil survey Ease of Online booking
fig = px.bar(df_ease_of_online_booking,
             x="index", 
             y="Ease of Online booking",
             title="Passenger Satisfaction on Ease of Online booking",
             labels={
                 "Ease of Online booking" : "Total Passengers",
                 "index": "Survey Results"
             },
             text="Ease of Online booking",
             category_orders={"index": [
                 "Netral", 
                 "Sangat Tidak Puas", 
                 "Tidak Puas", 
                 "Puas", 
                 "Lumayan Puas", 
                 "Sangat Puas"]})
fig.show()

# Menampilkan pie chart persentase hasil survey Ease of Online booking
fig = px.pie(df_ease_of_online_booking,
             names = "index", 
             values= "Ease of Online booking",
             title = "Percentage of Passenger Satisfaction on Ease of Online booking")
fig.show()

### Gate location

In [99]:
df_gate_location = pd.DataFrame(df['Gate location'].value_counts()).reset_index()

print('Hasil Survey Kepuasan Pada Layanan Gate location \n')
print(df_gate_location)

Hasil Survey Kepuasan Pada Layanan Gate location 

               index  Gate location
0  Puas               28577        
1  Lumayan Puas       24426        
2  Tidak Puas         19459        
3  Sangat Tidak Puas  17562        
4  Sangat Puas        13879        
5  Netral             1            


In [100]:
# Menampilkan barplot hasil survey Gate location
fig = px.bar(df_gate_location,
             x="index", 
             y="Gate location",
             title="Passenger Satisfaction on Gate location",
             labels={
                 "Gate location" : "Total Passengers",
                 "index": "Survey Results"
             },
             text="Gate location",
             category_orders={"index": [
                 "Netral", 
                 "Sangat Tidak Puas", 
                 "Tidak Puas", 
                 "Puas", 
                 "Lumayan Puas", 
                 "Sangat Puas"]})
fig.show()

# Menampilkan pie chart persentase hasil survey Gate location
fig = px.pie(df_gate_location,
             names = "index", 
             values= "Gate location",
             title = "Percentage of Passenger Satisfaction on Gate location")
fig.show()

### Food and drink

In [101]:
df_food_and_drink = pd.DataFrame(df['Food and drink'].value_counts()).reset_index()

print('Hasil Survey Kepuasan Pada Layanan Food and drink \n')
print(df_food_and_drink)

Hasil Survey Kepuasan Pada Layanan Food and drink 

               index  Food and drink
0  Lumayan Puas       24359         
1  Sangat Puas        22313         
2  Puas               22300         
3  Tidak Puas         21988         
4  Sangat Tidak Puas  12837         
5  Netral             107           


In [102]:
# Menampilkan barplot hasil survey Food and drink
fig = px.bar(df_food_and_drink,
             x="index", 
             y="Food and drink",
             title="Passenger Satisfaction on Food and drink",
             labels={
                 "Food and drink" : "Total Passengers",
                 "index": "Survey Results"
             },
             text="Food and drink",
             category_orders={"index": [
                 "Netral", 
                 "Sangat Tidak Puas", 
                 "Tidak Puas", 
                 "Puas", 
                 "Lumayan Puas", 
                 "Sangat Puas"]})
fig.show()

# Menampilkan pie chart persentase hasil survey Food and drink
fig = px.pie(df_food_and_drink,
             names = "index", 
             values= "Food and drink",
             title = "Percentage of Passenger Satisfaction on Food and drink")
fig.show()

### Online boarding

In [103]:
df_online_boarding = pd.DataFrame(df['Online boarding'].value_counts()).reset_index()

print('Hasil Survey Kepuasan Pada Layanan Online boarding \n')
print(df_inflight_wifi_service)

Hasil Survey Kepuasan Pada Layanan Online boarding 

               index  Inflight wifi service
0  Puas               25868                
1  Tidak Puas         25830                
2  Lumayan Puas       19794                
3  Sangat Tidak Puas  17840                
4  Sangat Puas        11469                
5  Netral             3103                 


In [104]:
# Menampilkan barplot hasil survey Online boarding
fig = px.bar(df_online_boarding,
             x="index", 
             y="Online boarding",
             title="Passenger Satisfaction on Online boarding",
             labels={
                 "Online boarding" : "Total Passengers",
                 "index": "Survey Results"
             },
             text="Online boarding",
             category_orders={"index": [
                 "Netral", 
                 "Sangat Tidak Puas", 
                 "Tidak Puas", 
                 "Puas", 
                 "Lumayan Puas", 
                 "Sangat Puas"]})
fig.show()

# Menampilkan pie chart persentase hasil survey Online boarding
fig = px.pie(df_online_boarding,
             names = "index", 
             values= "Online boarding",
             title = "Percentage of Passenger Satisfaction on Online boarding")
fig.show()

### Seat comfort

In [105]:
df_seat_comfort = pd.DataFrame(df['Seat comfort'].value_counts()).reset_index()

print('Hasil Survey Kepuasan Pada Layanan Seat comfort \n')
print(df_seat_comfort)

Hasil Survey Kepuasan Pada Layanan Seat comfort 

               index  Seat comfort
0  Lumayan Puas       31765       
1  Sangat Puas        26470       
2  Puas               18696       
3  Tidak Puas         14897       
4  Sangat Tidak Puas  12075       
5  Netral             1           


In [106]:
# Menampilkan barplot hasil survey Seat comfort
fig = px.bar(df_seat_comfort,
             x="index", 
             y="Seat comfort",
             title="Passenger Satisfaction on Seat comfort",
             labels={
                 "Seat comfort" : "Total Passengers",
                 "index": "Survey Results"
             },
             text="Seat comfort",
             category_orders={"index": [
                 "Netral", 
                 "Sangat Tidak Puas", 
                 "Tidak Puas", 
                 "Puas", 
                 "Lumayan Puas", 
                 "Sangat Puas"]})
fig.show()

# Menampilkan pie chart persentase hasil survey Seat comfort
fig = px.pie(df_seat_comfort,
             names = "index", 
             values= "Seat comfort",
             title = "Percentage of Passenger Satisfaction on Seat comfort")
fig.show()

### Inflight entertainment

In [107]:
df_inflight_entertainment = pd.DataFrame(df['Inflight entertainment'].value_counts()).reset_index()

print('Hasil Survey Kepuasan Pada Layanan Inflight entertainment \n')
print(df_inflight_entertainment)

Hasil Survey Kepuasan Pada Layanan Inflight entertainment 

               index  Inflight entertainment
0  Lumayan Puas       29423                 
1  Sangat Puas        25213                 
2  Puas               19139                 
3  Tidak Puas         17637                 
4  Sangat Tidak Puas  12478                 
5  Netral             14                    


In [108]:
# Menampilkan barplot hasil survey InflInflight entertainment
fig = px.bar(df_inflight_entertainment,
             x="index", 
             y="Inflight entertainment",
             title="Passenger Satisfaction on Inflight entertainment",
             labels={
                 "Inflight enteratiment" : "Total Passengers",
                 "index": "Survey Results"
             },
             text="Inflight entertainment",
             category_orders={"index": [
                 "Netral", 
                 "Sangat Tidak Puas", 
                 "Tidak Puas", 
                 "Puas", 
                 "Lumayan Puas", 
                 "Sangat Puas"]})
fig.show()

# Menampilkan pie chart persentase hasil survey Inflight entertainment
fig = px.pie(df_inflight_entertainment,
             names = "index", 
             values= "Inflight entertainment",
             title = "Percentage of Passenger Satisfaction on Inflight entertainment")
fig.show()

### On-board service

In [109]:
df_on_board_service = pd.DataFrame(df['On-board service'].value_counts()).reset_index()

print('Hasil Survey Kepuasan Pada Layanan On-board service \n')
print(df_on_board_service)

Hasil Survey Kepuasan Pada Layanan On-board service 

               index  On-board service
0  Lumayan Puas       30867           
1  Sangat Puas        23648           
2  Puas               22833           
3  Tidak Puas         14681           
4  Sangat Tidak Puas  11872           
5  Netral             3               


In [110]:
# Menampilkan barplot hasil survey On-board service
fig = px.bar(df_on_board_service,
             x="index", 
             y="On-board service",
             title="Passenger Satisfaction on On-board service",
             labels={
                 "On-board service" : "Total Passengers",
                 "index": "Survey Results"
             },
             text="On-board service",
             category_orders={"index": [
                 "Netral", 
                 "Sangat Tidak Puas", 
                 "Tidak Puas", 
                 "Puas", 
                 "Lumayan Puas", 
                 "Sangat Puas"]})
fig.show()

# Menampilkan pie chart persentase hasil survey On-board service
fig = px.pie(df_on_board_service,
             names = "index", 
             values= "On-board service",
             title = "Percentage of Passenger Satisfaction on On-board service")
fig.show()

### Leg room service

In [111]:
df_leg_room_service = pd.DataFrame(df['Leg room service'].value_counts()).reset_index()

print('Hasil Survey Kepuasan Pada Layanan Leg room service \n')
print(df_leg_room_service)

Hasil Survey Kepuasan Pada Layanan Leg room service 

               index  Leg room service
0  Lumayan Puas       28789           
1  Sangat Puas        24667           
2  Puas               20098           
3  Tidak Puas         19525           
4  Sangat Tidak Puas  10353           
5  Netral             472             


In [112]:
# Menampilkan barplot hasil survey Leg room service
fig = px.bar(df_leg_room_service,
             x="index", 
             y="Leg room service",
             title="Passenger Satisfaction on Leg room service",
             labels={
                 "Leg room service" : "Total Passengers",
                 "index": "Survey Results"
             },
             text="Leg room service",
             category_orders={"index": [
                 "Netral", 
                 "Sangat Tidak Puas", 
                 "Tidak Puas", 
                 "Puas", 
                 "Lumayan Puas", 
                 "Sangat Puas"]})
fig.show()

# Menampilkan pie chart persentase hasil survey Leg room service
fig = px.pie(df_leg_room_service,
             names = "index", 
             values= "Leg room service",
             title = "Percentage of Passenger Satisfaction on Leg room service")
fig.show()

### Baggage handling

In [113]:
df_baggage_handling = pd.DataFrame(df['Baggage handling'].value_counts()).reset_index()

print('Hasil Survey Kepuasan Pada Layanan Baggage handling \n')
print(df_baggage_handling)

Hasil Survey Kepuasan Pada Layanan Baggage handling 

               index  Baggage handling
0  Lumayan Puas       37383           
1  Sangat Puas        27131           
2  Puas               20632           
3  Tidak Puas         11521           
4  Sangat Tidak Puas  7237            


In [114]:
# Menampilkan barplot hasil survey Baggage handling
fig = px.bar(df_baggage_handling,
             x="index", 
             y="Baggage handling",
             title="Passenger Satisfaction on Baggage handling",
             labels={
                 "Baggage handling" : "Total Passengers",
                 "index": "Survey Results"
             },
             text="Baggage handling",
             category_orders={"index": [
                 "Netral", 
                 "Sangat Tidak Puas", 
                 "Tidak Puas", 
                 "Puas", 
                 "Lumayan Puas", 
                 "Sangat Puas"]})
fig.show()

# Menampilkan pie chart persentase hasil survey Baggage handling
fig = px.pie(df_baggage_handling,
             names = "index", 
             values= "Baggage handling",
             title = "Percentage of Passenger Satisfaction on Baggage handling")
fig.show()

### Checkin service

In [115]:
df_checkin_service = pd.DataFrame(df['Checkin service'].value_counts()).reset_index()

print('Hasil Survey Kepuasan Pada Layanan Checkin service \n')
print(df_checkin_service)

Hasil Survey Kepuasan Pada Layanan Checkin service 

               index  Checkin service
0  Lumayan Puas       29055          
1  Puas               28446          
2  Sangat Puas        20619          
3  Tidak Puas         12893          
4  Sangat Tidak Puas  12890          
5  Netral             1              


In [116]:
# Menampilkan barplot hasil survey Checkin service
fig = px.bar(df_checkin_service,
             x="index", 
             y="Checkin service",
             title="Passenger Satisfaction on Checkin service",
             labels={
                 "Checkin service" : "Total Passengers",
                 "index": "Survey Results"
             },
             text="Checkin service",
             category_orders={"index": [
                 "Netral", 
                 "Sangat Tidak Puas", 
                 "Tidak Puas", 
                 "Puas", 
                 "Lumayan Puas", 
                 "Sangat Puas"]})
fig.show()

# Menampilkan pie chart persentase hasil survey Checkin service
fig = px.pie(df_checkin_service,
             names = "index", 
             values= "Checkin service",
             title = "Percentage of Passenger Satisfaction on Checkin service")
fig.show()

### Inflight service

In [117]:
df_inflight_service = pd.DataFrame(df['Inflight service'].value_counts()).reset_index()

print('Hasil Survey Kepuasan Pada Layanan Inflight service \n')
print(df_inflight_service)

Hasil Survey Kepuasan Pada Layanan Inflight service 

               index  Inflight service
0  Lumayan Puas       37945           
1  Sangat Puas        27116           
2  Puas               20299           
3  Tidak Puas         11457           
4  Sangat Tidak Puas  7084            
5  Netral             3               


In [118]:
# Menampilkan barplot hasil survey Inflight service
fig = px.bar(df_inflight_service,
             x="index", 
             y="Inflight service",
             title="Passenger Satisfaction on Inflight service",
             labels={
                 "Inflight service" : "Total Passengers",
                 "index": "Survey Results"
             },
             text="Inflight service",
             category_orders={"index": [
                 "Netral", 
                 "Sangat Tidak Puas", 
                 "Tidak Puas", 
                 "Puas", 
                 "Lumayan Puas", 
                 "Sangat Puas"]})
fig.show()

# Menampilkan pie chart persentase hasil survey Inflight service
fig = px.pie(df_inflight_service,
             names = "index", 
             values= "Inflight service",
             title = "Percentage of Passenger Satisfaction on Inflight service")
fig.show()

### Cleanliness

In [119]:
df_cleanliness = pd.DataFrame(df['Cleanliness'].value_counts()).reset_index()

print('Hasil Survey Kepuasan Pada Layanan Cleanliness \n')
print(df_cleanliness)

Hasil Survey Kepuasan Pada Layanan Cleanliness 

               index  Cleanliness
0  Lumayan Puas       27179      
1  Puas               24574      
2  Sangat Puas        22689      
3  Tidak Puas         16132      
4  Sangat Tidak Puas  13318      
5  Netral             12         


In [120]:
# Menampilkan barplot hasil survey Cleanliness
fig = px.bar(df_cleanliness,
             x="index", 
             y="Cleanliness",
             title="Passenger Satisfaction on OnlCleanliness",
             labels={
                 "Cleanliness" : "Total Passengers",
                 "index": "Survey Results"
             },
             text="Cleanliness",
             category_orders={"index": [
                 "Netral", 
                 "Sangat Tidak Puas", 
                 "Tidak Puas", 
                 "Puas", 
                 "Lumayan Puas", 
                 "Sangat Puas"]})
fig.show()

# Menampilkan pie chart persentase hasil survey Cleanliness
fig = px.pie(df_cleanliness,
             names = "index", 
             values= "Cleanliness",
             title = "Percentage of Passenger Satisfaction on Cleanliness")
fig.show()

## Hasil Survey Secara Keseluruhan Terhadap Semua Layanan

In [121]:
fig = make_subplots()

# Diagram inflight wifi services
fig.add_trace(go.Bar(x=df_inflight_wifi_service["index"], 
                     y=df_inflight_wifi_service["Inflight wifi service"],
                     orientation='v',
                     name="Inflight wifi service",
                     text=df_inflight_wifi_service["Inflight wifi service"],
                     hoverinfo="all"))      

# Diagram Departure/Arrival time convenient
fig.add_trace(go.Bar(x=df_departure_arrival_time_convenient["index"], 
                     y=df_departure_arrival_time_convenient["Departure/Arrival time convenient"],
                     orientation='v',
                     name="Departure/Arrival time convenient",
                     text=df_departure_arrival_time_convenient["Departure/Arrival time convenient"],
                     hoverinfo="all"))

# Diagram Ease of Online booking
fig.add_trace(go.Bar(x=df_ease_of_online_booking["index"], 
                     y=df_ease_of_online_booking["Ease of Online booking"],
                     orientation='v',
                     name="Ease of Online booking",
                     text=df_ease_of_online_booking["Ease of Online booking"],
                     hoverinfo="all"))

# Diagram Gate location
fig.add_trace(go.Bar(x=df_gate_location["index"], 
                     y=df_gate_location["Gate location"],
                     orientation='v',
                     name="Gate location",
                     text=df_gate_location["Gate location"],
                     hoverinfo="all"))

# Diagram Food and drink
fig.add_trace(go.Bar(x=df_food_and_drink["index"], 
                     y=df_food_and_drink["Food and drink"],
                     orientation='v',
                     name="Food and drink",
                     text=df_food_and_drink["Food and drink"],
                     hoverinfo="all"))

# Diagram Online boarding
fig.add_trace(go.Bar(x=df_online_boarding["index"], 
                     y=df_online_boarding["Online boarding"],
                     orientation='v',
                     name="Online boarding",
                     text=df_online_boarding["Online boarding"],
                     hoverinfo="all"))

# Diagram Seat comfort
fig.add_trace(go.Bar(x=df_seat_comfort["index"], 
                     y=df_seat_comfort["Seat comfort"],
                     orientation='v',
                     name="Seat comfort",
                     text=df_seat_comfort["Seat comfort"],
                     hoverinfo="all"))

# Diagram Inflight Entertaiment
fig.add_trace(go.Bar(x=df_inflight_entertainment["index"], 
                     y=df_inflight_entertainment["Inflight entertainment"],
                     orientation='v',
                     name="Inflight entertainment",
                     text=df_inflight_entertainment["Inflight entertainment"],
                     hoverinfo="all"))

# Diagram On-board services
fig.add_trace(go.Bar(x=df_on_board_service["index"], 
                     y=df_on_board_service["On-board service"],
                     orientation='v',
                     name="On-board service",
                     text=df_on_board_service["On-board service"],
                     hoverinfo="all"))

# Diagram Leg room service
fig.add_trace(go.Bar(x=df_leg_room_service["index"], 
                     y=df_leg_room_service["Leg room service"],
                     orientation='v',
                     name="Leg room service",
                     text=df_leg_room_service["Leg room service"],
                     hoverinfo="all"))

# Diagram Baggage handling
fig.add_trace(go.Bar(x=df_baggage_handling["index"], 
                     y=df_baggage_handling["Baggage handling"],
                     orientation='v',
                     name="Baggage handling",
                     text=df_baggage_handling["Baggage handling"],
                     hoverinfo="all"))

# Diagram Checkin services
fig.add_trace(go.Bar(x=df_checkin_service["index"], 
                     y=df_checkin_service["Checkin service"],
                     orientation='v',
                     name="Checkin service",
                     text=df_checkin_service["Checkin service"],
                     hoverinfo="all"))

# Diagram Inflight services
fig.add_trace(go.Bar(x=df_inflight_service["index"], 
                     y=df_inflight_service["Inflight service"],
                     orientation='v',
                     name="Inflight service",
                     text=df_inflight_service["Inflight service"],
                     hoverinfo="all"))

# Diagram Cleanliness
fig.add_trace(go.Bar(x=df_cleanliness["index"], 
                     y=df_cleanliness["Cleanliness"],
                     orientation='v',
                     name="Cleanliness",
                     text=df_cleanliness["Cleanliness"],
                     hoverinfo="all"))

# Pemberian Judul, Legend, dan Susunan Survey
fig.update_layout(title_text="Survey Results For All Services", 
                  showlegend=True, 
                  xaxis=dict(title="Suvey Result", 
                             categoryorder="array", 
                             categoryarray=[
                                 "Netral", 
                                 "Sangat Tidak Puas", 
                                 "Tidak Puas", 
                                 "Puas", 
                                 "Lumayan Puas", 
                                 "Sangat Puas"]),
                  yaxis=dict(title="Total Passengers"))

fig.show()


# Kesimpulan dan Rekomendasi

## Kesimpulan

Berdasarkan data yang telah dianalisa, berikut ini adalah beberapa kesimpulan yang dapat ditemukan.

1. Berdasarkan hasil data, jumlah penumpang yang netral dan tidak puas adalah sebanyak 58.879 penumpang atau sebanyak 56.7% dari total keseluruhan penumpang.

1. Terjadi gap yang cukup besar antara penumpang yang merasa puas dan merasa tidak puas, pada tipe penumpang disloyal customer atau penumpang baru. 
Hal ini menandakan bahwa pelayanan yang diberikan dalam jangka waktu terdekat belakangan ini kurang memberikan kepuasan kepada penumpang terutama penumpang baru. 

1. Mayoritas penumpang yang merasa tidak puas adalah berusia antara 20 sampai 60 tahun, dengan jumlah tertinggi berada pada usia 25 dan 39 tahun.

1. Gap penumpang yang tidak puas juga terjadi pada penumpang dengan tujuan perjalanan personal travel.

1. Untuk kelas penerbangan, kelas penerbangan eco adalah kelas penerbangan yang paling memberikan pelayanan yang tidak memuaskan. 

1. Berdasarkan data dari hasil survey secara keseluruhan, berikut ini adalah **layanan yang harus segera ditingkatkan karena termasuk dalam 3 teratas hasil survey sangat tidak puas dan tidak puas:**
  1. Inflight wifi service:
    - Hasil survey sangat tidak puas: 17.840 atau 24.9% dari total keseluruhan penumpang.
    - Hasil survey tidak puas: 25.830 atau 24.9% dari total keseluruhan penumpang.
  1. Ease of Online booking:
    - Hasil survey sangat tidak puas: 17.525 atau 16.9% dari total keseluruhan penumpang.
    - Hasil survey tidak puas: 24.021 atau 23.1% dari total keseluruhan penumpang.

## Rekomendasi

Dari hasil analisa data yang telah dilakukan, berikut ini adalah beberapa rekomendasi yang dapat dilakukan untuk meningkatkan pelayanan maskapai penerbangan:

1. Pelayanan Inflight wifi service atau layanan wifi dalam pesawat dan ease of online booking atau kemudahan dalam melakukan pemesanan secara online adalah pelayanan yang harus segera ditingkatkan.
  - Rekomendasi bentuk pelayanan untuk meningkatkan kepuasan terhadap layanan Inflight wifi service:
    - Tingkatkan peralatan wifi di pesawat untuk memberikan koneksi internet yang lebih cepat dan stabil.
    - Berikan instruksi yang jelas tentang cara terhubung ke layanan wifi dan solusi masalah yang umum terjadi.
    - Tawarkan berbagai macam konten, seperti film, acara TV, dan musik, yang dapat diakses melalui layanan wifi di pesawat.
  - Rekomendasi bentuk pelayanan untuk meningkatkan kepuasan terhadap layanan ease of online booking
    - Sederhanakan proses pemesanan dengan mengurangi jumlah langkah dan klik yang diperlukan untuk menyelesaikan pemesanan.
    - Pastikan situs web atau aplikasi seluler mudah digunakan dan dapat diakses oleh semua jenis pengguna.
    - Tawarkan berbagai macam opsi pembayaran, termasuk dompet digital dan kartu kredit, untuk menyesuaikan dengan preferensi pelanggan yang berbeda.
    - Berikan ringkasan yang jelas dan singkat tentang detail pemesanan, termasuk jadwal perjalanan, biaya, dan syarat dan ketentuan, sebelum pelanggan mengkonfirmasi pemesanan.

1. Kedua pelayanan tersebut harus segera ditingkatkan karena berdasarkan data, mayoritas penumpang yang merasa tidak puas adalah penumpang baru.

1. Kedua layanan tersebut juga dapat ditingkatkan dengan menargetkan penumpang dengan usia pada rentang 20 sampai 60 tahun, atau lebih tepatnya untuk penumpang dengan usia 25 dan 39 tahun.

1. Dari 3 kelas penerbangan yang dimiliki saat ini, kelas penerbangan eco adalah kelas penerbangan yang harus diprioritaskan untuk ditingkatkan pelayanannya.

Save Data untuk Tableu

In [122]:
# df.to_csv('/content/drive/MyDrive/Purwadhika/Capstone Project Modul 2 - Eddy.csv', index=False)

In [123]:
df_survey_semua_layanan = df_inflight_wifi_service.merge(df_departure_arrival_time_convenient, on='index')\
.merge(df_ease_of_online_booking, on='index',how='outer')\
.merge(df_gate_location, on='index',how='outer')\
.merge(df_food_and_drink, on='index',how='outer')\
.merge(df_online_boarding, on='index',how='outer')\
.merge(df_seat_comfort, on='index',how='outer')\
.merge(df_inflight_entertainment, on='index',how='outer')\
.merge(df_on_board_service, on='index',how='outer')\
.merge(df_leg_room_service, on='index',how='outer')\
.merge(df_baggage_handling, on='index',how='outer')\
.merge(df_checkin_service, on='index',how='outer')\
.merge(df_inflight_service, on='index',how='outer')\
.merge(df_cleanliness, on='index',how='outer')

df_survey_semua_layanan['Baggage handling']=df_survey_semua_layanan.fillna(0)['Baggage handling'].astype(int)
df_survey_semua_layanan

# df_survey_semua_layanan.to_csv('/content/drive/MyDrive/Purwadhika/Hasil Survey Semua Layanan 2.csv', index=False)

index  Inflight wifi service  \
0  Puas               25868                   
1  Tidak Puas         25830                   
2  Lumayan Puas       19794                   
3  Sangat Tidak Puas  17840                   
4  Sangat Puas        11469                   
5  Netral             3103                    

   Departure/Arrival time convenient  Ease of Online booking  Gate location  \
0  17966                              24449                   28577           
1  17191                              24021                   19459           
2  25546                              19571                   24426           
3  15498                              17525                   17562           
4  22403                              13851                   13879           
5  5300                               4487                    1               

   Food and drink  Online boarding  Seat comfort  Inflight entertainment  \
0  22300           21804            18696         19139                    
1  21988           17505            14897         17637                    
2  24359           30762            31765         29423                    
3  12837           10692            12075         12478                    
4  22313           20713            26470         25213                    
5  107             2428             1             14                       

   On-board service  Leg room service  Baggage handling  Checkin service  \
0  22833             20098             20632             28446             
1  14681             19525             11521             12893             
2  30867             28789             37383             29055             
3  11872             10353             7237              12890             
4  23648             24667             27131             20619             
5  3                 472               0                 1                 

   Inflight service  Cleanliness  
0  20299             24574        
1  11457             16132        
2  37945             27179        
3  7084              13318        
4  27116             22689        
5  3                 12

In [124]:
df_survey_semua_layanan

index  Inflight wifi service  \
0  Puas               25868                   
1  Tidak Puas         25830                   
2  Lumayan Puas       19794                   
3  Sangat Tidak Puas  17840                   
4  Sangat Puas        11469                   
5  Netral             3103                    

   Departure/Arrival time convenient  Ease of Online booking  Gate location  \
0  17966                              24449                   28577           
1  17191                              24021                   19459           
2  25546                              19571                   24426           
3  15498                              17525                   17562           
4  22403                              13851                   13879           
5  5300                               4487                    1               

   Food and drink  Online boarding  Seat comfort  Inflight entertainment  \
0  22300           21804            18696         19139                    
1  21988           17505            14897         17637                    
2  24359           30762            31765         29423                    
3  12837           10692            12075         12478                    
4  22313           20713            26470         25213                    
5  107             2428             1             14                       

   On-board service  Leg room service  Baggage handling  Checkin service  \
0  22833             20098             20632             28446             
1  14681             19525             11521             12893             
2  30867             28789             37383             29055             
3  11872             10353             7237              12890             
4  23648             24667             27131             20619             
5  3                 472               0                 1                 

   Inflight service  Cleanliness  
0  20299             24574        
1  11457             16132        
2  37945             27179        
3  7084              13318        
4  27116             22689        
5  3                 12